### PR TITLE
fix: Use native focus-visible in addition to polyfill

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

We could potentially turn this off in production but kinda nice not to have to add an attribute and to be able to use devtools?

You can test this by going to the `Anchor` story page and opening devtools.  Select an Anchor element and use the pseudo state toggle to see it show up.  Then turn it off and add `data-focus-visible-added` and see it work again.

Screen Shot 2021-05-14 at 6 44 41 PM
Screen Shot 2021-05-14 at 6 45 04 PM

* Creates a new focus visible selector instead of replacing the native one with

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.05040e.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.05040e.0<br/>@codecademy/gamut-kit@0.3.103-alpha.05040e.0<br/>@codecademy/gamut-labs@17.1.1-alpha.05040e.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.05040e.0<br/>@codecademy/gamut-styles@10.0.1-alpha.05040e.0<br/>@codecademy/gamut-tests@2.3.31-alpha.05040e.0<br/>@codecademy/gamut@32.18.8-alpha.05040e.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.05040e.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.05040e.0<br/>@codecademy/styleguide@38.1.1-alpha.05040e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://609f0fc026ff1f0c30ab9a07--gamut-preview.netlify.app>https://609f0fc026ff1f0c30ab9a07--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/609f0fc026ff1f0c30ab9a07>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
> turn this off in production builds

Ah, yes please, can we? I'd rather not duplicate the selector in prod if we don't have to.

Otherwise strong +1 yes please to using the native dev tools! _shakes fist at Safari_

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.12.3-alpha.e9290a.0<br/>@codecademy/gamut-illustrations@0.14.4-alpha.e9290a.0<br/>@codecademy/gamut-kit@0.3.103-alpha.e9290a.0<br/>@codecademy/gamut-labs@17.1.1-alpha.e9290a.0<br/>@codecademy/gamut-patterns@0.1.4-alpha.e9290a.0<br/>@codecademy/gamut-styles@10.0.1-alpha.e9290a.0<br/>@codecademy/gamut-tests@2.3.31-alpha.e9290a.0<br/>@codecademy/gamut@32.18.8-alpha.e9290a.0<br/>@codecademy/konamimojisplosion@0.2.31-alpha.e9290a.0<br/>@codecademy/markdown-overrides@0.5.43-alpha.e9290a.0<br/>@codecademy/styleguide@38.1.1-alpha.e9290a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a28862cb640000b2403e77--gamut-preview.netlify.app>https://60a28862cb640000b2403e77--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a28862cb640000b2403e77>Deploy Logs</a></p>

**Review by dreamwasp — APPROVED** [review]
🚀

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60a836b8112911008cad0f6a--gamut-preview.netlify.app>https://60a836b8112911008cad0f6a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60a836b8112911008cad0f6a>Deploy Logs</a></p>

File: packages/gamut-styles/src/cache/stylisPlugins/focusVisible.ts [hunk 1]
```diff
@@ -2,9 +2,12 @@ import { StylisPlugin } from '@emotion/cache';
 
 export const focusVisible: StylisPlugin = (element) => {
   if (element.type === 'rule' && element.value.includes(':focus-visible')) {
-    element.props = element.props.map((prop) =>
-      prop.replace(/:focus-visible/g, '[data-focus-visible-added]')
-    );
+    element.props = element.props.map((prop) => {
+      const selector = process.env.NODE_ENV === 'dev' ? 'prop, ' : '';
```
Comment (JoshuaKGoldberg) [inline packages/gamut-styles/src/cache/stylisPlugins/focusVisible.ts:6]:
```suggestion
      const selector = process.env.NODE_ENV === 'dev' ? `${prop}, ` : '';
```

**Review by JoshuaKGoldberg — APPROVED** [review]
Yay!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.2-alpha.7bc2f8.0<br/>@codecademy/gamut-illustrations@0.15.2-alpha.7bc2f8.0<br/>@codecademy/gamut-kit@0.3.114-alpha.7bc2f8.0<br/>@codecademy/gamut-labs@18.1.5-alpha.7bc2f8.0<br/>@codecademy/gamut-patterns@0.2.2-alpha.7bc2f8.0<br/>@codecademy/gamut-styles@10.1.2-alpha.7bc2f8.0<br/>@codecademy/gamut-tests@2.4.2-alpha.7bc2f8.0<br/>@codecademy/gamut@33.1.1-alpha.7bc2f8.0<br/>@codecademy/konamimojisplosion@0.2.33-alpha.7bc2f8.0<br/>@codecademy/markdown-overrides@0.5.45-alpha.7bc2f8.0<br/>@codecademy/styleguide@40.0.2-alpha.7bc2f8.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@4.14.2-alpha.9524ae.0<br/>@codecademy/gamut-illustrations@0.16.1-alpha.9524ae.0<br/>@codecademy/gamut-kit@0.3.116-alpha.9524ae.0<br/>@codecademy/gamut-labs@18.1.7-alpha.9524ae.0<br/>@codecademy/gamut-patterns@0.2.2-alpha.9524ae.0<br/>@codecademy/gamut-styles@10.1.2-alpha.9524ae.0<br/>@codecademy/gamut-tests@2.4.2-alpha.9524ae.0<br/>@codecademy/gamut@33.1.3-alpha.9524ae.0<br/>@codecademy/konamimojisplosion@0.2.33-alpha.9524ae.0<br/>@codecademy/markdown-overrides@0.5.45-alpha.9524ae.0<br/>@codecademy/styleguide@40.0.4-alpha.9524ae.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60ad82091ae44111283774ec--gamut-preview.netlify.app>https://60ad82091ae44111283774ec--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60ad82091ae44111283774ec>Deploy Logs</a></p>

**Pull request merged**